src/pages/signup-affiliates/components/_account-personal-details.tsx

```diff
@@ -0,0 +1,333 @@
+        switch (name) {
+            case 'username': {
+                setUsername(value)
+                return setUserNameErrorMsg(affiliate_validation.username(value))
+            }
+            case 'first_name': {
+                setFirstName(value)
+                return setFirstNameErrorMsg(affiliate_validation.first_name(value))
+            }
+            case 'last_name': {
+                setLastName(value)
+                return setLastNameErrorMsg(affiliate_validation.last_name(value))
+            }
+            case 'date_birth': {
+                setDateBirth(value)
+                break
+            }
+            case 'company_name': {
+                setCompanyName(value)
+                return setCompanyNameErrorMsg(affiliate_validation.company_name(value))
+            }
+            case 'company_registration_number': {
+                setCompanyRegistrationNumber(value)
+                return setCompanyRegistrationErrorMsg(
+                    affiliate_validation.company_registration_number(value),
+                )
+            }
+            case 'website_url': {
+                setWebsiteUrl(value)
+                return setWebsiteUrlErrorMsg(affiliate_validation.website_url(value))
+            }
+            case 'social_media_url': {
+                setSocialMedia(value)
+                return setSocialMediaErrorMsg(affiliate_validation.social_media_url(value))
+            }
+            case 'password': {
+                setPassword(value)
+                return setPasswordErrorMsg(affiliate_validation.password(value))
+            }
+        }
+    }
+
+    return (
+        <InputGroup>
+            <InputWrapper>
+                <form>
+                    {getFormFields().map((item) => {
+                        if (item.name === 'date_birth') {
+                            return (
+                                <BirthForm
+                                    id={item.id}
+                                    key={item.id}
+                                    error={item.error}
+                                    value={item.value}
+                                    label={localize(item.label)}
+                                    setFieldValue={item.value_set}
+                                />
+                            )
+                        } else {
+                            return (
+                                <AffiliateInput
+                                    id={item.id}
+                                    key={item.id}
+                                    name={item.name}
+                                    type={item.type}
+                                    value={item.value}
+                                    error={item.error}
+                                    required={item.required}
+                                    extra_info={item.extra_info}
+                                    label={localize(item.label)}
```

Comment:
seems like we are already localizing labels in `account-personal-details.tsx` do we need to do it here also?

Reply:
right! good catch